src/utils.ts

```diff
@@ -285,6 +291,68 @@ export async function getBrowserPath(config: Partial<IUserConfig> = {}): Promise
     }
 }
 
+/**
+ * Gets a supported debug config and updates the status of the launch.json file associated with the current workspace
+ * @returns {vscode.DebugConfiguration | null}
+ */
+ export function getLaunchJson(): vscode.DebugConfiguration | null {
+    // Check if there is a folder open
+    if (!vscode.workspace.workspaceFolders) {
+        void vscode.commands.executeCommand('setContext', 'launchJsonStatus', 'None');
+        return null;
+    }
+
+    // Check if there's a launch.json file
+    const workspaceUri = vscode.workspace.workspaceFolders[0].uri;
+    const filePath = `${workspaceUri.fsPath}/.vscode/launch.json`;
+    if (fse.pathExistsSync(filePath)) {
+        // Check if there is a supported debug config
+        const configs = vscode.workspace.getConfiguration('launch', workspaceUri).get('configurations') as vscode.DebugConfiguration[];
+        for (const config of configs) {
+            if (config.type === 'vscode-edge-devtools.debug' || config.type === 'msedge' || config.type === 'edge') {
+                void vscode.commands.executeCommand('setContext', 'launchJsonStatus', 'Supported');
+                return config;
+            }
+        }
+        void vscode.commands.executeCommand('setContext', 'launchJsonStatus', 'Unsupported');
+        return null;
+    }
+        void vscode.commands.executeCommand('setContext', 'launchJsonStatus', 'None');
+        return null;
+
+}
+
+/**
+ * Add a template for a supported debug configuration to launch.json
+ * @returns {void}
+ */
+export async function configureLaunchJson(): Promise<void> {
+    if (!vscode.workspace.workspaceFolders)
```

Comment:
I wonder if we should find a way to handle this case more gracefully for users. Would they ever be in a scenario where they try to create a launchJson without opening a workspace folder? would they get an error message if this fails or would it just be silent? I could see the silent fail case being confusing for users if they hit it.

Reply:
I don't think that scenario should be possible because in package.json, the button to create a launch.json is only shown if there is a workspace folder open. The if statement was needed to quiet a typescript error about `workspaceFolders` possibly being undefined. We'll need to be careful of that if we use this function in more places later on, so I guess we might as well put in an error message now.